miniserv.pl

```diff
@@ -1399,12 +1399,19 @@ sub handle_request
 			}
 		local $url = $wantport == 443 ? "https://$urlhost/"
 					      : "https://$urlhost:$wantport/";
+		local $jsurl = $config{'musthost'} ?
+				                   $url :
+				                   "https://'+location.host+'";
+		local $jsredir = $config{'musthost'} ?
+				                   "location.href='$url'" :
+				                   "location.protocol='https:'";
 		&http_error(200, "Document follows",
 			"This web server is running in SSL mode. ".
-			"Try the URL <a href='$url'>$url</a> instead.".
+			"Trying to redirect to <a href='$url'>$url</a> instead ...".
 			"<script>".
 			"if (location.protocol != 'https:') {".
-			"  location.protocol = 'https:';".
+			"  document.querySelector('a').href='".$jsurl."';document.querySelector('a').innerText='".$jsurl."';".
```

Comment:
Why the need to update the URL using javascript when we just printed it above?

Reply:
Well, if the host in the URL is forced with `musthost` (which 99% of a time isn't), then it's all fine.

However, as you mentioned earlier yourself (that host is not always detected correctly) it may print a wrong host on the redirect message, but JavaScript using `location.protocol='https:` will redirect it to whatever is in initial query.

i.e. the message can be:

<img width="815" alt="image" src="https://user-images.githubusercontent.com/4426533/190874645-08a4b541-0b39-49b4-95b2-b800447ac13c.png">


.. and the final redirect done by JavaScript may happen to the other URL, as Webmin cannot detect host correctly.

I found this very confusing and solved this problem with this PR. So, if we say that redirect is happening to `alma9-pro.synology:10000` we cannot eventually redirect to `mail.alma9-pro.synology:10000` ..

Reply:
Ok I see now ... thanks!

Reply:
Anything else needed to merge it?